txpool/txpool.go

```diff
@@ -133,10 +139,67 @@ func (p *txPool) checkTx(trx *tx.Tx) error {
 	return nil
 }
 
-func (p *txPool) RemoveTx(id tx.ID) {
+func (p *txPool) HandleCommittedBlock(blk *block.Block) error {
 	p.lk.Lock()
 	defer p.lk.Unlock()
 
+	for _, trx := range blk.Transactions() {
+		p.removeTx(trx.ID())
+
+		p.handleIncreaseConsumption(trx)
+	}
+
+	return p.handleDecreaseConsumption(blk.Height())
+}
+
+func (p *txPool) handleIncreaseConsumption(trx *tx.Tx) {
+	if trx.IsTransferTx() || trx.IsBondTx() || trx.IsWithdrawTx() {
+		signer := trx.Payload().Signer()
+
+		p.consumptionMap[signer] += uint32(trx.SerializeSize())
+	}
```

Comment:
I think it's better to have `atomic` increase to prevent race conditions on `consumptionMap`

Reply:
Here don't use unsafe memory, no need to use atomic.

Reply:
For the consumptionMap map[crypto.Address]uint32, atomic operations are typically not necessary unless multiple goroutines concurrently read and modify the same uint32 values in the map. A map in Go is not thread-safe, so accessing or modifying it concurrently requires synchronization (e.g., using sync.Mutex).

However, if the map is safely guarded by a lock or accessed in a single-threaded context, atomic operations for the values aren't required.

Reply:
I mean, we can't guarantee scope of use of `consumptionMap` in long term.
Although we can guarantee that `handleIncreaseConsumption` is locked but it is likely that another function or goroutine accesses `consumptionMap` in different scope using different lock, which is not safe.

Reply:
`handleIncreaseConsumption` and `handleDecreaseConsumption` have parent lock from `HandleCommittedBlock`.

In HandleCommittedBlock we don't have concurrent for increase and decrease.